kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -794,133 +683,98 @@ define([
+                        body: fileLinksPanelBody,
+                    })
+                );
             }
-
             events.attachEvents();
-
             this.loading(false);
+            return Promise.all(renderPromises);
         },
         loading: function (isLoading) {
             if (isLoading) {
-                this.showMessage('<i class="fa fa-spinner fa-spin"></i>');
+                const span = html.tag('span'),
+                    i = html.tag('i'),
+                    loadingMsg = span(
+                        i({
+                            class: 'fa fa-spinner fa-spin',
+                        })
+                    );
+                this.$messagePane.append(loadingMsg);
+                this.$messagePane.show();
             } else {
-                this.hideMessage();
+                this.$messagePane.hide();
+                this.$messagePane.empty();
             }
         },
-        showMessage: function (message) {
-            const span = $('<span/>').append(message);
-            this.$messagePane.append(span);
-            this.$messagePane.show();
-        },
-        hideMessage: function () {
-            this.$messagePane.hide();
-            this.$messagePane.empty();
-        },
-        clientError: function (error) {
+        /**
+         * This shows an error from either a string or an object, wiping out everything else in this
+         * widget's body.
+         * The common KBase JSON-RPC error object format is:
+         * {
+         *   error: {
+         *     error: 'some error message',
+         *     ... other keys ...
+         *   },
+         *   status: 500
+         * }
+         * though this is possible, too:
+         * {
+         *   error: {
+         *     message: 'some message'
+         *   }
+         * }
+         * @param {Object | String} error
+         */
+        showClientError: function (error) {
             this.loading(false);
             let errString = 'Unknown error.';
-            console.error(error);
+            console.error(JSON.stringify(error));
+            // if we get a basic error as a string
             if (typeof error === 'string') {
                 errString = error;
+                // this can be thrown from certain network error cases
             } else if (error.error && error.error.message) {
                 errString = error.error.message;
+                // this mess can come from a workspace API error
             } else if (error.error && error.error.error && typeof error.error.error === 'string') {
                 errString = error.error.error;
             }
 
-            const $errorDiv = $('<div>')
-                .addClass('alert alert-danger')
-                .append('<b>Error:</b>')
-                .append('<br>' + errString);
-            this.$elem.empty();
-            this.$elem.append($errorDiv);
+            const div = html.tag('div'),
+                b = html.tag('b'),
```

Comment:
do we have the `strong` html tag? that's more semantically correct than using `b`

Reply:
Yup, updated that too.